### PR TITLE
createst: add midstream arg - v5

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: https://github.com/OISF/suricata-verify/pull/800

Changes from previous PR:
- rebased
- added `min-version` option explanation to the README file

### CODE REVIEW

File: createst.py [hunk 1]
```diff
@@ -155,6 +160,21 @@ def check_requires():
         if args[item]:
             return True
 
+def check_set_args():
+    """
+    Check if the user wants midstream set to true and/or to have strict
+    checksums
+    """
+    user_args = ["strictcsums", "midstream"]
+    no_user_args = True
```
Comment (inashivb) [inline createst.py:169]: Can we make it work without this extra variable?

File: createst.py [hunk 2]
```diff
@@ -155,6 +160,21 @@ def check_requires():
...
+    for item in user_args:
+        if args[item]:
+            if item != "strictcsums":
+                return True
+            no_user_args = False
+        elif no_user_args:
+            return True
```
Comment (inashivb) [inline createst.py:176]: We can probably clean this part a bit then.

**Review by inashivb — COMMENTED** [review]
Great job! :) Sorry for taking so long to get back. I think it works well now but you can clean the code a bit. Not marking it as changes requested but if you'd like to give cleaning the code a try, go ahead. :)

**Comment by inashivb** [comment]
Continuing our discussion here: does it make sense for us to have a `no-strictcsums` instead of what we have right now? to avoid the complexity that this is bringing in the code.. But, how nice is that from a user's perspective?

**Comment by jufajardini** [comment]
Feedback incorporated in https://github.com/OISF/suricata-verify/pull/851

**Pull request closed without merging**